src/PathProcessor/CivicrmPathProcessor.php

```diff
@@ -11,42 +11,39 @@
  */
 class CivicrmPathProcessor implements InboundPathProcessorInterface {
 
+  private static $knownPaths;
+
   /**
    * {@inheritdoc}
    */
   public function processInbound($path, Request $request) {
     // If the path is a civicrm path.
     if (strpos($path, '/civicrm/') === 0) {
-      // Initialize civicrm.
-      $civicrm = new Civicrm();
-      $civicrm->initialize();
-      // Fetch civicrm menu items.
-      $items = \CRM_Core_Menu::items();
-      $longest = '';
-      foreach (array_keys($items) as $item) {
-        $item = '/' . $item;
-        // If he current path is a civicrm path.
-        if ((strpos($path, $item) === 0)) {
-          // Discover longest matching civicrm path in the request path.
-          if (strlen($item) > strlen($longest)) {
-            $longest = $item;
-          }
+      // We've already looked up this path on this page request.
+      if (!isset(self::$knownPaths[$path])) {
+        // Initialize civicrm.
+        $civicrm = new Civicrm();
+        $civicrm->initialize();
+        $routesInitialized = \CRM_Core_DAO::singleValueQuery('SELECT COUNT(*) from civicrm_menu');
+        if (!$routesInitialized) {
+          \CRM_Core_Menu::store();
         }
-      }
-      if (!empty($longest)) {
-        // Parse url component parameters from path.
-        $params = str_replace($longest, '', $path);
-        // Replace slashes with colons and the controller will piece it back
-        // together.
-        if (strlen($params)) {
-          $params = str_replace('/', ':', $params);
-          if (substr($params, 0, 1) == ':') {
-            $params = substr($params, 1);
+        $pathArray = \CRM_Core_Menu::get(ltrim($path, '/'));
+        // Restore the leading slash.
+        if ($pathArray['path'] ?? FALSE) {
+          $newPath = '/' . $pathArray['path'];
+          // Parse url component parameters from path.
+          $params = str_replace($newPath, '', $path);
+          // Replace slashes with colons and the controller will piece it back
+          // together.
+          if (strlen($params)) {
+            $params = str_replace('/', ':', $params);
+            if (substr($params, 0, 1) == ':') {
+              $params = substr($params, 1);
+            }
+            $newPath .= "/$params";
           }
-          return "$longest/$params";
-        }
-        else {
-          return $longest;
+          self::$knownPaths[$newPath] = TRUE;
```

Comment:
It sets this here, but then never uses it again except to check if it's set, and then down below ALWAYS returns $path. Is it missing a `return $newPath` here?

Reply:
Oh - hmm, yes.  I need to revise this.  I did my initial tests against pages that didn't have params (i.e. the route matched `civicrm_menu` exactly).  Then I modified it to support params but didn't go back and revise this.

Reply:
Also...when did this get to be 5 levels of nested `if`s? Yeesh.

Reply:
OK - I fixed this and reduced the nested `if`s to a max of 3, but I haven't tested the new patch yet.  I was actually hoping some CI existed on this repo that would run tests, but oh well.  I'll look at this later today.

Reply:
> 5 levels of nested ifs

That's how you know it's civi. No copyright block needed in the file.

> hoping some CI existed on this repo

There isn't anything automatic, but what prompted me to take a closer look at the code was that as a sanity check I ran it with civicarrot against the cdntaxreceipts extension's tests. You managed to crash github so that there aren't even logs of the phpunit part, which is no easy task: https://github.com/SemperIT/CiviCARROT/actions/runs/6476564520/job/17585470546. It's running again now: https://github.com/SemperIT/CiviCARROT/actions/runs/6485761313/job/17612539157

civicrm_entity and webform's tests also run in a drupal 8 environment, but I don't think they hit too many civi urls, usually just on the drupal side, so for this PR might not pick anything up.

Reply:
It crashed again but not as bad. Normally the run takes about 15 min so I'm guessing it's getting into some loop. Just clicking around manually I don't see anything obvious. I'll try to see what the test issue is about (the config file warning isn't relevant).